pac4j-core/src/main/java/org/pac4j/core/client/Clients.java

```diff
@@ -73,28 +74,32 @@ public Clients(final Client... clients) {
     }
 
     /**
-     * Initialize all clients by computing callback urls.
+     * Initialize all clients by computing callback urls if necessary.
      */
     @Override
     protected void internalInit() {
         CommonHelper.assertNotNull("clients", this.clients);
-        if (CommonHelper.isNotBlank(this.callbackUrl)) {
-            for (final Client client : this.clients) {
-                if (client instanceof IndirectClient) {
-                    final IndirectClient indirectClient = (IndirectClient) client;
-                    String indirectClientCallbackUrl = indirectClient.getCallbackUrl();
-                    // no callback url defined for the client -> set it with the group callback url
-                    if (indirectClientCallbackUrl == null) {
-                        indirectClient.setCallbackUrl(this.callbackUrl);
-                        indirectClientCallbackUrl = this.callbackUrl;
-                    }
-                    // if the "clientName" parameter is not already part of the callback url, add it unless the client
-                    // has indicated to not include it.
-                    if (indirectClient.isIncludeClientNameInCallbackUrl() &&
-                            indirectClientCallbackUrl.indexOf(this.clientNameParameter + "=") < 0) {
-                        indirectClient.setCallbackUrl(CommonHelper.addParameter(indirectClientCallbackUrl, this.clientNameParameter,
-                                indirectClient.getName()));
-                    }
+        final HashSet<String> names = new HashSet<>();
+        for (final Client client : this.clients) {
+            final String name = client.getName();
+            if (names.contains(name)) {
```

Comment:
The `Clients` is built from a callback url and a list of clients, but I fear that users may have twice the same clients (and indeed the same names) without noticing. At the time of creation, I don't see how the user would check the names (which are not defined, but inferred from the clients).
Example: https://github.com/pac4j/j2e-pac4j-demo/blob/master/src/main/java/org/pac4j/demo/j2e/config/DemoConfigFactory.java#L27